spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -261,7 +268,10 @@ protected InputFile getInputFile(String location) {
 
     @Override
     protected void markRowDeleted(InternalRow row) {
-      row.setBoolean(columnIsDeletedPosition(), true);
+      if (!row.getBoolean(columnIsDeletedPosition())) {
+        row.setBoolean(columnIsDeletedPosition(), true);
+        counter().increment();
+      }
```

Comment:
`markRowDeleted` can be called as the `Consumer` in `Deletes.markDeleted(CloseableIterable<T>, Predicate<T>, Consumer<T>)` via `DeleteFilter#applyPosDeletes(CloseableIterable<T>)` and via `DeleteFilter#applyEqDeletes(CloseableIterable<T>)`.
I assume we want to avoid counting twice when the same row is deleted via a positional delete and then via an equality delete.
Thus we check to see if the row is already marked deleted, and if it is not, then we mark it and increment the count.
This count is tested in `TestSparkReaderDeletes#testMixedPosAndEqDeletesWithDeletedColumn`.